src/relay/op/tensor/transform.cc

```diff
@@ -1076,7 +1076,7 @@ Examples::
     .set_support_level(3)
     .add_type_rel("Take", TakeRel)
     .set_attr<FTVMCompute>("FTVMCompute", TakeCompute)
-    .set_attr<TOpPattern>("TOpPattern", kInjective);
+    .set_attr<TOpPattern>("TOpPattern", kOpaque);
```

Comment:
I hope we could come up with a way to avoid this. This would hurt performance on hummingbird workload and possibly other models.

Reply:
For example, maybe we can use `annotation.stop_fusion` when we encounter take + dynamic, solving this problem at the frontend.

Reply:
Good thing I have a typo in CI.

I'm not sure I see a clean way to do this in the frontends, it demands we already have infer_type run to check for dynamic inputs.

Maybe we write a pass that selectively stops fusion on certain ops under certain conditions?

Reply:
Yes a new pass that inserts `stop_fusion` sounds good. I can work on this. We can make `take` opaque for now until I have that new pass ready.

Anyway I think graph runtime shouldn't be affected by the issue of `take` + dynamic, so `take` should be injective eventually.

Reply:
@mbrookhart Can you add `TODO(masahi)` there?

Reply:
@masahi want to request changes so this doesn't merge while we chat about it?

Reply:
sure done. But I think it is good to go after adding a comment on why we make this change temporary

Reply:
I think the other option is to edit the relay take function that creates the op. We could remove the normalization that causes this problem from the take kernel in topi, and do in it relay with select/shape_of, but that might end up causing some performance degradation, it's hard to predict.